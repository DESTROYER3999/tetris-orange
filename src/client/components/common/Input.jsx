import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";

const useStyles = makeStyles(theme => ({
  textField: {
    marginLeft: theme.spacing(1),
    marginRight: theme.spacing(1)
  },
  input: {
    fontFamily: "ArcadeClassic"
  },
  label: {
    fontFamily: "ArcadeClassic"
  }
}));

const Input = React.forwardRef((props, ref) => {
  const classes = useStyles();
  return (
    <TextField
      {...props}
      inputRef={ref}
      className={classes.textField}
      margin="normal"
      variant="outlined"
      InputProps={{ classes: { root: classes.input } }}
      InputLabelProps={{ classes: { root: classes.label } }}
    />
  );
});

export default Input;
