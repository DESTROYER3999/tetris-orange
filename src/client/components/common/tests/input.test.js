import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Input from "../Input";

describe("<Input />", () => {
  test("Input rendering without crashing", () => {
    const wrapper = shallow(<Input />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
