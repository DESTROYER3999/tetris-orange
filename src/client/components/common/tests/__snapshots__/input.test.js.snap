// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Input rendering without crashing 1`] = `
<WithStyles(ForwardRef(TextField))
  InputLabelProps={
    Object {
      "classes": Object {
        "root": "makeStyles-label-3",
      },
    }
  }
  InputProps={
    Object {
      "classes": Object {
        "root": "makeStyles-input-2",
      },
    }
  }
  className="makeStyles-textField-1"
  inputRef={null}
  margin="normal"
  variant="outlined"
/>
`;
